engine/runtime/src/main/scala/org/enso/compiler/pass/resolve/TypeNames.scala

```diff
@@ -77,24 +77,29 @@ case object TypeNames extends IRPass {
     expression.transformExpressions {
       case expr if SuspendedArguments.representsSuspended(expr) => expr
       case n: IR.Name.Literal =>
-        bindingsMap
-          .resolveName(n.name)
-          .map(res => n.updateMetadata(this -->> Resolution(res)))
-          .fold(
-            error =>
-              IR.Error.Resolution(n, IR.Error.Resolution.ResolverError(error)),
-            n =>
-              n.getMetadata(this).get.target match {
-                case _: ResolvedModule =>
-                  IR.Error.Resolution(
-                    n,
-                    IR.Error.Resolution.UnexpectedModule("type signature")
-                  )
-                case _ => n
-              }
-          )
+        resolveName(bindingsMap, n)
+      case n: IR.Name.Qualified =>
+        resolveName(bindingsMap, n)
     }
 
+  private def resolveName(bindingsMap: BindingsMap, name: IR.Name): IR.Name =
```

Comment:
Probably insufficient for [ascribedWithAParameter test](https://github.com/enso-org/enso/pull/6584/files#diff-0ee393cab082d78e4b3dbe155decce832501c8cb080b9bbfb6961940dbfc7a08R92).